File: 1 - Arrays/1 - findPair/2 - findPair.java
// To use sort
import java.util.Arrays;

class findPair
{
	public static void findPair(int arr[], int sum)
	{
		Arrays.sort(arr);
		int low = 0;
		int high = arr.length-1; 

		while (low<high)
		{
			if (arr[low]+arr[high]==sum)
			{
				System.out.println("Pair found: " + arr[low] + " and " + arr[high]);
			}
			if (arr[low]+arr[high]<sum)
				low++;
			else
				high--;
		}
		System.out.println("Pair not found");
	}

	public static void main(String[] args)
	{
		int arr[] = {8,7,2,5,3,1};
		int sum = 10;
		findPair(arr, sum);
	}
}

File: 1 - Arrays/1 - findPair/2 - findPair.cpp
// Notes link: http://www.geeksforgeeks.org/c-qsort-vs-c-sort/
//   Will use cpp for STL's sort, rather than C's qsort.


#include <stdio.h>
#include <stdlib.h>

void findPair(int arr[], int n, int sum);

int main()
{
	int arr[] = {8,7,2,5,3,1};
	int sum = 10;

	int n = sizeof(arr)/sizeof(arr[0]);

	findPair(arr, n, sum);

	return 0;
}

void findPair(int arr[], int n, int sum)
{
	std::sort(arr, arr+n);
	int low = 0;
	int high = n-1;

	while (low<high)
	{
		if (arr[low] + arr[high] == sum)
		{
			std::cout << "Pair found" << endl;
		}
		(arr[low] + arr[high] < sum)? low++: high--;
	}

	std::cout << "Pair not found" << endl;
}